tests/test_util/test_warning.py

```diff
@@ -0,0 +1,149 @@
+
+    expected = changed(a=5, b=3)
+    got = changed(x=5, y=3)
+    assert got == expected
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning")
+def test_function_deprecated_args_docstring():
+    @deprecated_args(
+        [("x", "a"), ("y", "b")],
+        version="a future release",
+        reason="x and y are not descriptive",
+    )
+    def changed(a: int, b: int) -> int:
+        """This is a test function"""
+        return a + b
+
+    assert changed.__doc__ == "This is a test function"
+
+
+@pytest.mark.parametrize("kwargs", (dict(x=5, y=3), dict(a=5, y=3), dict(x=5, b=3)))
+def test_function_deprecated_args_warns(kwargs):
+    # Example target function to be decorated
+    @deprecated_args(
+        [("x", "a"), ("y", "b")],
+        version="a future release",
+        reason="x and y are not descriptive",
+    )
+    def changed(a: int, b: int) -> int:
+        return a + b
+
+    with pytest.deprecated_call():
+        changed(**kwargs)
+
+
+def test_function_redundant_deprecated_args_does_not_warn():
+    @deprecated_args(
+        [("x", "a"), ("y", "b")],
+        version="a future release",
+        reason="x and y are not descriptive",
+    )
+    def changed(a: int, b: int) -> int:
+        return a + b
+
+    with pytest.warns(None) as record:  # verify no warnings for correct parameters
+        changed(a=5, b=3)
+    assert not record
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning")
+def test_function_deprecated_args_pickled():
+    @deprecated_args(
+        [("x", "a"), ("y", "b")],
+        version="a future release",
+        reason="x and y are not descriptive",
+    )
+    def changed(a: int, b: int) -> int:
+        return a + b
+
+    myfunc = changed(x=1, y=2)
+    pickled_func = pickle.dumps(myfunc)
+    assert type(pickled_func) == bytes
+    unpickled_func = pickle.loads(pickled_func)
+    assert unpickled_func == changed(a=1, b=2)
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning")
+def test_method_deprecated_args():
+    class foo:
+        @deprecated_args(
+            [("x", "a"), ("y", "b")],
+            version="a future release",
+            reason="x and y are not descriptive",
+        )
+        def changed(self, a: int, b: int) -> int:
+            return a + b
+
+    expected = foo().changed(a=5, b=3)
+    got = foo().changed(x=5, y=3)
+    assert got == expected
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning")
+def test_method_deprecated_args_docstring():
+    class foo:
+        @deprecated_args(
+            [("x", "a"), ("y", "b")],
+            version="a future release",
+            reason="x and y are not descriptive",
+        )
+        def changed(self, a: int, b: int) -> int:
+            """This is a test function"""
+            return a + b
+
+    assert foo().changed.__doc__ == "This is a test function"
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning")
+def test_function_deprecated_args_pickled():
```

Comment:
3 copies of this same test name

Reply:
I think I addressed the text over-rides comment